core/src/main/java/org/apache/calcite/sql/fun/PostgreSQLCastOperator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.fun;
+
+import org.apache.calcite.sql.SqlBinaryOperator;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.SqlSyntax;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.type.InferTypes;
+import org.apache.calcite.sql.validate.SqlMonotonicity;
+
+/**
+ * PostgreSQL's casting operator <code>::</code>
+ *
+ */
+public class PostgreSQLCastOperator extends SqlBinaryOperator {
+  /**
```

Comment:
You have 2 choice:
1. Inherit `SqlSpecialOperator` instead of `SqlBinaryOperator `
2. Use SqlCastFunction() directly in the parser but customize the unparse sql call logic[1]

I suggest you chose 2.

[1]https://github.com/apache/calcite/blob/1748f0503e7b626a8d0165f1698adb8b61bbc31e/core/src/main/java/org/apache/calcite/sql/SqlDialect.java#L400

Reply:
Why are you suggesting to use a `SqlSpecialOperator` ?

Reply:
Let's choose 2 please, make a new operator is not necessary, the reason i suggest `SqlSpecialOperator` is because the parse block where you add the PG_CAST grammar all use this Operator, and the `SqlSpecialOperator` doc says that this kind of operator is used for special grammar.